main/network.c

```diff
@@ -430,30 +430,27 @@ php_socket_t php_network_bind_socket_to_local_addr(const char *host, unsigned po
 	for (sal = psal; *sal != NULL; sal++) {
 		sa = *sal;
 
-		/* create a socket for this address */
-		sock = socket(sa->sa_family, socktype, 0);
-
-		if (sock == SOCK_ERR) {
-			continue;
-		}
-
 		switch (sa->sa_family) {
 #if HAVE_GETADDRINFO && HAVE_IPV6
 			case AF_INET6:
-				((struct sockaddr_in6 *)sa)->sin6_family = sa->sa_family;
 				((struct sockaddr_in6 *)sa)->sin6_port = htons(port);
 				socklen = sizeof(struct sockaddr_in6);
 				break;
 #endif
 			case AF_INET:
-				((struct sockaddr_in *)sa)->sin_family = sa->sa_family;
 				((struct sockaddr_in *)sa)->sin_port = htons(port);
 				socklen = sizeof(struct sockaddr_in);
 				break;
 			default:
-				/* Unknown family */
-				socklen = 0;
-				sa = NULL;
```

Comment:
if `sa` is NULL, it will close the socket and create a new socket in the next loop. We can check it first, then we create the socket.